### ISSUE TITLE
[BUG] Breadcrumb Not Rendering Dynamic Segments, e.g. :projectId

### ISSUE THREAD

**Issue by aress31** [refinedev/refine#6584]
### Describe the bug

The breadcrumb component is not rendering dynamic segments. See the screenshot below:

![image](https://github.com/user-attachments/assets/e6b9367b-4c76-441b-9cf1-9fb05b56aead)

### Steps To Reproduce

N/A

### Expected behavior

Should render dynamic segment.

### Packages

All to latest.

**Comment by Samuel-Habib** [comment]
Just to be clear, by breadcrumb components you mean the Projects / Applications part? What specific dynamic segments are you referring to?

**Comment by Samuel-Habib** [comment]
I also wanted to clarify as to how you got to the projects route, did you make that route yourself or load it from somewhere else because i don't see it listed as one of the routes in the repo.

**Comment by aress31** [comment]
> Just to be clear, by breadcrumb components you mean the Projects / Applications part? What specific dynamic segments are you referring to?

That is correct I mean the `Projects / Application` part - the missing part is the `Projects / :projectId / Applications` as shown in the URL in the screenshot. See the resources:

```js
  createResource("dashboard", {
    meta: {
      hide: true,
      parent: "projects",
    },
  }),
  createResource("applications", {
    meta: {
      --- SNIP ---
      hide: true,
      parent: "projects"
    },
  }),
```

In short, the <Breadcrumb> don't seem to parse dynamic segments like `:projectId` and the documentation has no information about that and there is no example with dynamic path - because it is not supported.

**Comment by aress31** [comment]
@alicanerdurmaz exactly what I want and what I would expect the breadcrumb to do. :)

Is there a way to make it work as it is now? Haven't found a solution in the doc and digging the code, it seems that this is not supported as it is not parsing url params to replace placeholders like :projectId.

**Comment by alicanerdurmaz** [comment]
There is no way to do that right now. I wrote hard-coded code to explain the feature.

We need to implement this on [useBreadcrumb hook](https://github.com/refinedev/refine/blob/main/packages/core/src/hooks/breadcrumb/index.ts#L33). 

Our hands are full with other tasks, but we're always open to PRs 🙌

**Comment by darshan-mundra-08** [comment]
Is there a data stored in relation to the unique uuid in the URL? Eg: some readable name to be shown in the breadcrumb area? from where we can extract the readable name from the uuid and show it there in the breadcrumb

**Comment by aress31** [comment]
@darshan-mundra-08 should be define in resource routes? Can we make parent accept strings like ":projectId"?

**Comment by darshan-mundra-08** [comment]
@aress31 No, Its not important that it should be in the source resource routes? the data should be defined anywhere in the project whose variable are accessible to the breadcrumb component.

**Comment by ShashankGupta10** [comment]
Can I work on this guys?

**Comment by alicanerdurmaz** [comment]
Hello, @darshan-mundra-08 or @aress31 Do you want to work on this? If you don't I will assigned to @ShashankGupta10

**Comment by darshan-mundra-08** [comment]
@alicanerdurmaz  I would love to contribute to it if you want.

**Comment by aress31** [comment]
@alicanerdurmaz feel free to assign @ShashankGupta10 or @darshan-mundra-08 for this ticket.

**Comment by alicanerdurmaz** [comment]
I assigned to @darshan-mundra-08 because @darshan-mundra-08 asked first. Thank you for improving Refine 🚀

**Comment by darshan-mundra-08** [comment]
Thanks @alicanerdurmaz @aress31
